src/main/scala/edu/ie3/util/osm/model/OsmEntity.scala

```diff
@@ -0,0 +1,153 @@
+import edu.ie3.util.osm.model.OsmEntity.Relation.RelationMember
+import org.locationtech.jts.geom.Point
+
+import java.time.Instant
+import scala.annotation.tailrec
+
+sealed trait OsmEntity {
+  val id: Long
+  val tags: Map[String, String]
+  val metaInformation: Option[MetaInformation]
+
+  def hasKeyValuePair(key: CommonOsmKey, value: String): Boolean =
+    hasKeyValuePair(key.toString, value)
+
+  def hasKeyValuePair(key: String, value: String): Boolean =
+    tags.get(key).contains(value)
+
+  def hasKeyValuesPairOr(key: CommonOsmKey, values: Set[String]): Boolean =
+    hasKeyValuesPairOr(key.toString, values)
+
+  /** values.isEmpty equals to tagName=*
+    *
+    * @param key
+    * @param values
+    * @return
+    */
+  def hasKeyValuesPairOr(key: String, values: Set[String]): Boolean =
+    tags.get(key) match {
+      case Some(tagValue) if values.nonEmpty =>
+        values.contains(tagValue)
+      case Some(tagValue) =>
+        true
+      case None => false
+    }
+
+  @tailrec
+  final def hasKeysValuesPairOr(
+      keyTagValues: Map[String, Set[String]]
+  ): Boolean = {
+    keyTagValues.headOption match {
+      case Some((key, tagValues)) =>
+        // if we have at least one matching key/tagValues pair
+        // we can stop the process and return true
+        if (hasKeyValuesPairOr(key, tagValues)) {
+          true
+        } else {
+          hasKeysValuesPairOr(keyTagValues - key)
+        }
+      case None =>
+        false
+    }
+  }
+
+  def hasKey(key: CommonOsmKey): Boolean =
+    hasKeyValuesPairOr(key, Set.empty)
+
+  def hasKey(key: String): Boolean =
+    hasKeyValuesPairOr(key, Set.empty)
+
+}
+
+object OsmEntity {
+
+  final case class MetaInformation(
+      version: Option[Int] = None,
+      timestamp: Option[Instant] = None,
+      changeSet: Option[Long] = None,
+      userId: Option[Int] = None,
+      userName: Option[String] = None,
+      visible: Option[Boolean] = None
+  )
+
+  final case class Node(
+      override val id: Long,
+      latitude: Double,
+      longitude: Double,
+      override val tags: Map[String, String],
+      override val metaInformation: Option[MetaInformation] = None
+  ) extends OsmEntity {
+    val coordinate: Point = GeoUtils.xyToPoint(longitude, latitude)
+  }
+
+  sealed trait Way extends OsmEntity {
+    val id: Long
+    val nodes: Seq[Long]
+    val tags: Map[String, String]
+    val metaInformation: Option[MetaInformation]
+  }
+
+  object Way {
+
+    final case class OpenWay private[model] (
+        override val id: Long,
+        override val nodes: Seq[Long],
+        override val tags: Map[String, String],
+        override val metaInformation: Option[MetaInformation]
+    ) extends Way
+
+    final case class ClosedWay private[model] (
+        override val id: Long,
+        override val nodes: Seq[Long],
+        override val tags: Map[String, String],
+        override val metaInformation: Option[MetaInformation]
+    ) extends Way
```

Comment:
Please add the following methods without implementation (logic will be brought by #175)

`ClosedWay#area: Try[ComparableQuantity[Area]]` and `ClosedWay#center: Coordinate` or as an alternative to `Coordinate` from jts also some light-weight class that is defined by us (might have a convenience method to convert to jts Coordinate).